### PR TITLE
Update README to include zstd workaround for windows.

### PR DESCRIPTION
**Pull request by pdotl**
This PR contains the following changes:

1. Move `Cache segment restore timeout` section to `Know practices and workarounds`
2. Add `yaml` formatting for actions code blocks
3. Add zstd workaround for windows and zstd requirement for self-hosted windows runners.

### CODE REVIEW

**Comment by wrexbe** [comment]
This doesn't fix it. Even in your own test, this doesn't fix it
https://github.com/pdotl/sample-cache-demo/runs/7959974880?check_suite_focus=true
The Cache Python virtualinv step on Ubuntu took 2 seconds, and on Windows it took 11 seconds.
That is more then 5 times slower on windows, even with a tiny cache.

**Comment by pdotl** [comment]
Hi @wrexbe, I agree. This workaround is being used by customers vs using `gzip` which is even slower. We are simply documenting the same. We are looking into improving windows performance for cache by looking at several alternatives.

File: README.md [hunk 1]
```diff
@@ -223,18 +223,18 @@ jobs:
         if: steps.cache-primes.outputs.cache-hit != 'true'
         run: ./generate-primes -d prime-numbers
 ```
-## Cache segment restore timeout
 
+## Known practices and workarounds
```
Comment (Phantsure) [inline README.md:227]: Would it make sense to create separate file for this?

Comment (pdotl) [reply README.md:227]: I think it would yes. We can move it to another file.

**Review by Phantsure — APPROVED** [review]
LGTM

**Pull request merged**